app/controllers/users/onboardings_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Users
+  class OnboardingsController < ApplicationController
+    before_action :set_no_cache_header
+    before_action :authenticate_user!
+    after_action :verify_authorized
+
+    ALLOWED_USER_PARAMS = %i[last_onboarding_page username].freeze
+    ALLOWED_ONBOARDING_PARAMS = %i[checked_code_of_conduct checked_terms_and_conditions].freeze
+
+    def onboarding_update
+      authorize User
+
+      user_params = {}
+
+      if params[:user]
+        if params.dig(:user, :username).blank?
```

Comment:
Main has this check:

```ruby
      if params[:user].key?(:username) && params[:user][:username].blank?
```

The reason for this is so that the patch requests to onboarding_update don't return a 422 error on each slide except the one where we set the username (and bio).

Reply:
Ah, I understand!